src/Query/Query.php

```diff
@@ -385,6 +387,19 @@ public function filterWhere(array $condition): static
         return $this;
     }
 
+    public function for(ExpressionInterface|string|null $value = null): static
+    {
+        $this->for = $value;
+        return $this;
+    }
+
+    public function forUpdate(bool $value = true): static
+    {
+        return $this->for(
+            $value ? new Expression('UPDATE') : null
+        );
```

Comment:
```suggestion
    public function forUpdate(): static
    {
        return $this->for('UPDATE');
```
`bool $value = true` seems redundant parameter.

Use `forUpdate()` for setting `FOR UPDATE` clause and `for()` to reset it.

Reply:
I assumed this usage:

```php
$userConditionForUseUpdate = ...;
$query
    ->from(...)
    ->where(...)
    ->andWhere(...)
    ->forUpdate($userConditionForUseUpdate);
```

Do you think it's not need?

Reply:
Could it be

```php
$userForUpdate = ... ? 'UPDATE' : null;
$query
    ->from(...)
    ->where(...)
    ->andWhere(...)
    ->for($userForUpdate);
```

Reply:
Agree. I'll fix it.

Reply:
done